File: Orimath/Controls/PolygonControl.cs
using System.Windows;
using System.Windows.Controls;
using System.Windows.Shapes;

namespace Orimath.Controls
{
    public class PolygonControl : Control
    {
        public double Points { get => (double)GetValue(PointsProperty); set => SetValue(PointsProperty, value); }
        public static readonly DependencyProperty PointsProperty = Polygon.PointsProperty.AddOwner(typeof(PolygonControl));

        static PolygonControl()
        {
            DefaultStyleKeyProperty.OverrideMetadata(
                typeof(PolygonControl),
                new FrameworkPropertyMetadata(typeof(PolygonControl)));
        }
    }
}

File: Plugins/Orimath.Basics.View/Controls/PolygonControl.fs
namespace Orimath.Basics.View.Controls
open System.Windows
open System.Windows.Controls
open System.Windows.Media
open System.Windows.Shapes

type PolygonControl() =
    inherit Control()
    static let mutable init = false
    do
        if not init then
            init <- true
            FrameworkElement.DefaultStyleKeyProperty.OverrideMetadata(
                typeof<PolygonControl>, FrameworkPropertyMetadata(typeof<PolygonControl>))

    member this.Points
        with get() = this.GetValue(PolygonControl.PointsProperty) :?> PointCollection
        and set (v: PointCollection) = this.SetValue(PolygonControl.PointsProperty, box v)
    static member val PointsProperty = Polygon.PointsProperty.AddOwner(typeof<PolygonControl>)
